src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -186,52 +211,143 @@ dumpBabelGUCs(Archive *fout)
 }
 
 /*
- * bbf_selectDumpableCast: Mark a cast as to be dumped or not
+ * bbf_selectDumpableObject:
+ *		Mark a generic dumpable object as to be dumped or not
  */
 void
-bbf_selectDumpableCast(CastInfo *cast)
+bbf_selectDumpableObject(DumpableObject *dobj, Archive *fout)
 {
-	TypeInfo      *sTypeInfo;
-	TypeInfo      *tTypeInfo;
-	ExtensionInfo *ext = findOwningExtension(cast->dobj.catId);
-
-	/* Skip if cast is not a member of babelfish extension */
-	if (ext == NULL || strcmp(ext->dobj.name, "babelfishpg_common") != 0)
+	if (!isBabelfishDatabase(fout))
 		return;
 
-	sTypeInfo = findTypeByOid(cast->castsource);
-	tTypeInfo = findTypeByOid(cast->casttarget);
-
-	/*
-	 * Do not dump following unused CASTS:
-	 * pg_catalog.bool -> sys.bpchar
-	 * pg_catalog.bool -> sys.varchar
-	 */
-	if (sTypeInfo && tTypeInfo &&
-			sTypeInfo->dobj.namespace &&
-			tTypeInfo->dobj.namespace &&
-			strcmp(sTypeInfo->dobj.namespace->dobj.name, "pg_catalog") == 0 &&
-			strcmp(tTypeInfo->dobj.namespace->dobj.name, "sys") == 0 &&
-			strcmp(sTypeInfo->dobj.name, "bool") == 0 &&
-			(strcmp(tTypeInfo->dobj.name, "bpchar") == 0 ||
-			 strcmp(tTypeInfo->dobj.name, "varchar") == 0))
-		cast->dobj.dump = DUMP_COMPONENT_NONE;
-}
+	switch (dobj->objType)
+	{
+		case DO_CAST:
+			{
+				CastInfo    	*cast = (CastInfo *) dobj;
+				TypeInfo    	*sTypeInfo;
+				TypeInfo    	*tTypeInfo;
+				ExtensionInfo	*ext = findOwningExtension(cast->dobj.catId);
+
+				/* Skip if cast is not a member of babelfish extension */
+				if (ext == NULL || strcmp(ext->dobj.name, "babelfishpg_common") != 0)
+					return;
+
+				sTypeInfo = findTypeByOid(cast->castsource);
+				tTypeInfo = findTypeByOid(cast->casttarget);
+
+				/*
+				* Do not dump following unused CASTS:
+				* pg_catalog.bool -> sys.bpchar
+				* pg_catalog.bool -> sys.varchar
+				*/
+				if (sTypeInfo && tTypeInfo &&
+						sTypeInfo->dobj.namespace &&
+						tTypeInfo->dobj.namespace &&
+						strcmp(sTypeInfo->dobj.namespace->dobj.name, "pg_catalog") == 0 &&
+						strcmp(tTypeInfo->dobj.namespace->dobj.name, "sys") == 0 &&
+						strcmp(sTypeInfo->dobj.name, "bool") == 0 &&
+						(strcmp(tTypeInfo->dobj.name, "bpchar") == 0 ||
+						strcmp(tTypeInfo->dobj.name, "varchar") == 0))
+					cast->dobj.dump = DUMP_COMPONENT_NONE;
+			}
+			break;
+		case DO_TABLE:
+			{
+				TableInfo *tbinfo = (TableInfo *) dobj;
+
+				if (fout->dopt->binary_upgrade)
+					return;
```

Comment:
Do we need this condition for CAST object as well?

Reply:
Nope, this check to exclude MVU scenario but CAST check is needed for both MVU and dump/restore.